packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -142,18 +143,25 @@ export default async function ShadowSystem(world: World) {
   let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, 0)
   dropShadows.matrixAutoUpdate = false
 
-  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function modifyShadowCount() {
+  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function (props) {
+    if (getShadowsEnabled()) return null
+
     world.scene.remove(dropShadows)
```

Comment:
logic here should be in a `useEffect(() => ` with the dependencies being whatever this relies on (drop shadow radius, group component length etc)